libsolidity/interface/CompilerStack.cpp

```diff
@@ -1022,22 +1022,29 @@ string CompilerStack::createMetadata(Contract const& _contract) const
 
 bytes CompilerStack::createCBORMetadata(string const& _metadata, bool _experimentalMode)
 {
+	bytes const& versionBytes = VersionCompactBytes;
+	unsigned versionBytesSize = versionBytes.size();
+	solAssert(versionBytesSize <= 0xff, "");
 	bytes cborEncodedHash =
-		// CBOR-encoding of the key "bzzr0"
-		bytes{0x65, 'b', 'z', 'z', 'r', '0'}+
-		// CBOR-encoding of the hash
-		bytes{0x58, 0x20} + dev::swarmHash(_metadata).asBytes();
+		// CBOR-encoding of the key "bzzr0" (TextString)
+		bytes{0x65, 'b', 'z', 'z', 'r', '0'} +
+		// CBOR-encoding of the hash (ByteString)
+		bytes{0x58, 0x20} + dev::swarmHash(_metadata).asBytes() +
+		// CBOR-encoding of the key "solc" (TextString)
+		bytes{0x64, 's', 'o', 'l', 'c'} +
+		// CBOR-encoding of the version string (ByteString)
+		((versionBytesSize <= 23) ? bytes{static_cast<unsigned char>(0x40 + versionBytesSize)} : bytes{0x58, static_cast<unsigned char>(versionBytesSize)}) + versionBytes;
 	bytes cborEncodedMetadata;
 	if (_experimentalMode)
 		cborEncodedMetadata =
 			// CBOR-encoding of {"bzzr0": dev::swarmHash(metadata), "experimental": true}
-			bytes{0xa2} +
+			bytes{0xa3} +
 			cborEncodedHash +
 			bytes{0x6c, 'e', 'x', 'p', 'e', 'r', 'i', 'm', 'e', 'n', 't', 'a', 'l', 0xf5};
```

Comment:
This is rediculous. Let's just add the 10 (?) additional bytes for the version string.

I see that it includes the commit hash. Do we really need that?